esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -898,6 +898,28 @@ void WsWuInfo::getGraphTimingData(IArrayOf<IConstECLTimingData> &timingData)
         it->playProperties(visitor);
 }
 
+void WsWuInfo::getServiceNames(IEspECLWorkunit &info, unsigned long flags)
+{
+    if (!(flags & WUINFO_IncludeServiceNames))
+        return;
+
+    StringArray serviceNames;
+    WuScopeFilter filter("depth[1,]");
+    filter.sourceFlags = SSFsearchGraph;
+    filter.addOutputAttribute(WaServiceName);
+    Owned<IConstWUScopeIterator> it = &cw->getScopeIterator(filter);
+    ForEach(*it)
+    {
+        if (it->getScopeType() != SSTactivity)
+            continue;
+
+        StringBuffer serviceName;
+        it->queryAttribute(WaServiceName, serviceName);
```

Comment:
This needs to be

    const char * value = it->queryAttribute(WaServiceName, serviceName);
    if (value)

There is no guarantee that the parameter passed as a scratchpad will be modfied.

Reply:
Will change